### PR TITLE
dev-lang/ghc: keyword 9.0.2 for ~amd64, ~ppc64, ~x86

### PR DESCRIPTION
**Pull request by matoro**
This brings up much of the Haskell ecosystem to 9.0.x, AND introduces a port for ppc64le.  More ports to follow.

Unfortunately I was not able to bootstrap any of the existing keyworded ghc versions.  9.0.2 was the earliest version that I was able to bootstrap using Debian's binaries (also at 9.0.2).

To make sure that the ecosystem functioned reasonably with ghc 9, and on ppc64le, I keyworded the dep trees for app-text/pandoc and dev-util/shellcheck.  This covers a good chunk of dev-haskell/*.  **All unit tests (where present) passed for all packages included here.**

Unfortunately due to the tendency for Haskell packages to impose tight upper bound checks, along with a major ghc version bump, this required upgrades for probably most of the dep trees.  However, rest assured that I **did not create any of the new ebuilds added here, neither by hand nor with hackport.  ALL of the new ebuilds are taken verbatim from https://github.com/gentoo-haskell/gentoo-haskell.  Where ::haskell had the same version but with patches or adjusted dependency checks, I revbumped the version in ::gentoo even though not strictly required, and linked the associated ::haskell commit in the commit message.**

There is one exception to the sentence above:  I tweaked =app-text/pandoc-2.18 from a local `lua53` USE flag to use `lua-single.eclass`.

Let me know if there's anything I can do to make this easier to review.

### CODE REVIEW

**Comment by matoro** [comment]
Put this as draft temporarily to shut up the bot, but it should be ready.

**Comment by mgorny** [comment]
@matoro, I think I've found the problem — we didn't protect against running the script multiple times simultaneously, and due to connectivity issues multiple instances ended up running in parallel — then they tried to process the same PR simultaneously.

**Comment by mgorny** [comment]
The next assignment run should clean up the duplicate comments. That is, when it finally manages to connect ;-).

**Comment by thesamesam** [comment]
CC @ceamac, @hololeap, @solpeth as FYI (I don't expect comment necessarily)

File: dev-haskell/cabal/cabal-3.6.3.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	HCFLAGS="${HCFLAGS} -ignore-package=pulseaudio"
+	HCFLAGS="${HCFLAGS} -ignore-package=kinds"
+}
+
+src_configure() {
+	haskell-cabal_src_configure \
+		--flag=-bundled-binary-generic
+}
+
+CABAL_CORE_LIB_GHC_PV="PM:8.10.3 PM:8.10.4 PM:8.10.4-r1 PM:9999"
```
Comment (solpeth) [inline dev-haskell/cabal/cabal-3.6.3.0.ebuild:55]: This version of `cabal` is not bundled with ghc-9.0.2, so this won't work. We need to keep bundled libraries versioned exactly the same as what they are in ghc. For an idea of this, do `git grep 'CABAL_CORE_LIB_GHC_PV'` in the gentoo-haskell repo and it'll show you which packages are bundled with ghc-9.0.2. Happy to help with this over the next few days if need be.

File: dev-haskell/binary/binary-0.8.7.0.ebuild [hunk 2]
```diff
@@ -18,7 +18,7 @@ SRC_URI="https://hackage.haskell.org/package/${P}/${P}.tar.gz"
 LICENSE="BSD"
 SLOT="0/${PV}"
 # keep in sync with ghc-8.8
```
Comment (solpeth) [inline dev-haskell/binary/binary-0.8.7.0.ebuild:20]: We need version 0.8.8.0 to be compatible with ghc-9.0.2. All in `::haskell` ready to go, so you should be able to pull it over to here with only minor changes if any.

File: dev-haskell/text/text-1.2.4.1_rc1.ebuild [hunk 3]
```diff
@@ -22,7 +22,7 @@ SRC_URI="https://github.com/haskell/${PN}/archive/v${MY_PV}.tar.gz -> ${P}.tar.g
 LICENSE="BSD-2"
 SLOT="0/${PV}"
 # keep in sync with ghc-8.10.3
```
Comment (solpeth) [inline dev-haskell/text/text-1.2.4.1_rc1.ebuild:24]: We need `text-1.2.5.0` brought over from `::haskell` with the correct `CABAL_CORE_LIB_GHC_PV` line.

**Review by solpeth — CHANGES_REQUESTED** [review]
We also need `haskeline-0.8.2.0` and `terminfo-0.4.1.5` brought over from `::haskell` with the appropriate `CABAL_CORE_LIB_GHC_PV=PM:9.0.2` line, and we need to add `PM:9.0.2` to xhtml if it is not already there.

**Comment by solpeth** [comment]
I don't know if you wanted to tackle this now or later, but it would be good to bring in the latest `hackport` from `::haskell` as well (if you haven't already - I can't see it here).

We also have eclass changes in `::haskell` that should probably be brought in here if they haven't already. But @hololeap has more of a handle on that.

**Review by solpeth — APPROVED** [review]
I've looked at most of the changes directly - they all look fine. Bundled libs changes should be fine. In isolation, each change seems good to go.

I have not tested this batch of changes collectively on a live system or container with only the ::gentoo repo, but assuming the PR author has done this then no problems should arise.

N.B.: there are a number of major version bumps in here. This is why we need to ensure there are no dependency conflicts between packages, which is why testing on a container or live system is important to not break ::gentoo's haskell packages.

**Pull request closed without merging**